virtwho/config.py

```diff
@@ -1093,9 +1094,40 @@ def _validate_owner(self, key):
 
     def _validate_filter(self, filter_key):
         """
-        Try to validate filter option
+        Try to validate filter option. It can contain list of hostname and UUIDs
+        :param list_key: key of filter ('filter_hosts', 'exclude_hosts')
+        :return: None or list of warnings
         """
-        return self._validate_list(filter_key)
+
+        # Check validity of list first
+        result = self._validate_list(filter_key)
+        if result is not None:
+            # When validation of list failed, then there is no reason for
+            # further validation, because self._values[filter_key] is empty
+            return result
+
+        hypervisor_id = self._values.get('hypervisor_id')
+
+        # When hypervisor_id is not 'uuid' (it is hostname or hwuuid), then
+        # filter should not contain any UUID, because no UUIDs will be gathered
+        # Note: example of hwuuid is e.g. 'host-9' (it doesn't look like UUID)
+        if hypervisor_id != 'uuid':
+            wrong_filter_values = []
+            for filter_value in self._values[filter_key]:
+                try:
+                    uuid.UUID(filter_value)
+                except ValueError:
+                    pass
+                else:
+                    wrong_filter_values.append(filter_value)
+            if len(wrong_filter_values) > 0:
+                result = (
+                    'warning',
+                    'Filter values: "%s" are not compatible with hypervisor_id = "%s"' %
```

Comment:
I think this warning could be more informative... Perhaps something like

```
Filter values: "%s" appear to be UUIDs. UUIDs are not gathered when hypervisor_id = "%s"
```

@cnsnyder thoughts?

Reply:
I agree with this suggestion.

Reply:
Sounds good.